plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcRecordCursor.java

```diff
@@ -227,6 +228,13 @@ public void close()
         try (Connection connection = this.connection;
                 Statement statement = this.statement;
                 ResultSet resultSet = this.resultSet) {
+            if (statement != null) {
+                try {
+                    statement.cancel();
```

Comment:
Add a comment why. Note that we call statement.close() in a moment, so it's unclear why additional cancel helps.